backend/src/grynos.js

```diff
@@ -26,16 +27,42 @@ const mapCourseFromGrynos = (course) => ({
     location: course.location,
 });
 
-const fetchCourses = async () => {
+const mapCourseDetailsFromGrynos = async (course) => {
+    const courseDetails = await axios(courseDetailUrl + course.code);
+    return {
+        ...course,
+        description: courseDetails.data.description,
+        single_payment_count: courseDetails.data.singlePaymentCount,
+        company_name: courseDetails.data.companyName,
+        course_type_id: courseDetails.data.courseTypeID,
+        course_type_name: courseDetails.data.courseTypeName,
+        teacher: courseDetails.data.teacher,
+    };
+};
+
+const fetchCoursesFromGrynos = async () => {
     const response = await axios(url);
     if (response.data.course) {
-        return response.data.course.map(mapCourseFromGrynos);
+        return await Promise.all(
+            response.data.course
+                .map(mapCourseFromGrynos)
+                .map(mapCourseDetailsFromGrynos)
+        );
     }
 };
 
 const updateCoursesToDb = async () => {
     try {
-        const courses = await fetchCourses();
+        await sequelize.sync();
+        let courses = await fetchCoursesFromGrynos();
+        const dbCourses = await db.courses.getCourses();
+        if (dbCourses.length > 0) {
+            courses = courses.filter((course) => {
+                return dbCourses.filter(
+                    (dbCourse) => course.id !== dbCourse.id
+                );
```

Comment:
This is now just returning true every time as the `.filter` returns a new array. Use something like `.find` instead or compare the length of filter result.